tests/mongo-go-driver.yml

```diff
@@ -87,8 +87,8 @@ results:
   ferretdb:
     stats:
       expected_fail: 10
-      expected_skip: 19
-      expected_pass: 633
```

Comment:
Why the number of passing tests went down?

Reply:
I think it was caused by the `mongo` submodule being out of sync with shell dependency, #564 should fix this.

Reply:
That's needs some investigation.

Reply:
We seem to be running less subtests in total with the update. This is _not_ the case for MongoDB. I am unsure why this is happening and I took a lot at some of the [commits](https://github.com/mongodb/mongo-go-driver/compare/6d814184973ae7ffc13281c93b423dc522a4bcca...6c7e1241b747d2fb5d164c9ca9d17617f33e7899) for possible causes and could not find any. I logged some more output to verify that:

```
FerretDB:

CURRENT
commit: 6d814184973ae7ffc13281c93b423dc522a4bcca
Expectedly ignored: 16
Total number of tests: 678

UPDATE
commit: 6c7e1241b747d2fb5d164c9ca9d17617f33e7899
Expectedly ignored: 15
Total number of tests: 662
```

Any ideas?